kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/MMBearerIpFamilyTest.java

```diff
@@ -0,0 +1,194 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Eurotech and/or its affiliates and others
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *  Eurotech
+ *******************************************************************************/
+package org.eclipse.kura.nm;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Set;
+
+import org.eclipse.kura.net.status.modem.BearerIpType;
+import org.freedesktop.dbus.types.UInt32;
+import org.junit.Test;
+import org.junit.experimental.runners.Enclosed;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Enclosed.class)
+public class MMBearerIpFamilyTest {
+
+    @RunWith(Parameterized.class)
+    public static class MMBearerIpFamilyToMMBearerIpFamilyTest {
+
+        @Parameters
+        public static Collection<Object[]> BearerIpFamilyParams() {
+            List<Object[]> params = new ArrayList<>();
+            params.add(new Object[] { new UInt32(0x00000000L), MMBearerIpFamily.MM_BEARER_IP_FAMILY_NONE });
+            params.add(new Object[] { new UInt32(0x00000001L), MMBearerIpFamily.MM_BEARER_IP_FAMILY_IPV4 });
+            params.add(new Object[] { new UInt32(0x00000002L), MMBearerIpFamily.MM_BEARER_IP_FAMILY_IPV6 });
+            params.add(new Object[] { new UInt32(0x00000004L), MMBearerIpFamily.MM_BEARER_IP_FAMILY_IPV4V6 });
+            params.add(new Object[] { new UInt32(0x00000008L), MMBearerIpFamily.MM_BEARER_IP_FAMILY_NON_IP });
+            params.add(new Object[] { new UInt32(0xFFFFFFF7L), MMBearerIpFamily.MM_BEARER_IP_FAMILY_ANY });
+            params.add(new Object[] { new UInt32(0x12345678L), MMBearerIpFamily.MM_BEARER_IP_FAMILY_NONE });
+            return params;
+        }
+
+        private final UInt32 inputIntValue;
+        private final MMBearerIpFamily expectedIpFamily;
+        private MMBearerIpFamily calculatedIpFamily;
+
+        public MMBearerIpFamilyToMMBearerIpFamilyTest(UInt32 intValue, MMBearerIpFamily ipFamily) {
+            this.inputIntValue = intValue;
+            this.expectedIpFamily = ipFamily;
+        }
+
+        @Test
+        public void shouldReturnCorrectIpFamily() {
+            whenCalculateMMBearerIpFamily();
+            thenCalculatedMMBearerIpFamilyIsCorrect();
+        }
```

Comment:
Personally I prefer to see the input and output of the unit tests as parameters of the `given`/`when`/`then` methods to emphasise what we're testing:

```suggestion
        public void shouldReturnCorrectIpFamily() {
            whenConverterIsCalledWith(this.inputIntValue);
            thenCalculatedMMBearerIpFamilyEquals(this.expectedIpFamily);
        }
```

This is less relevant here since the test is parametrised, but it might be easier for debugging since it's perfectly clear where the input and expected values are stored.